packages/flutter/lib/src/widgets/gesture_detector.dart

```diff
@@ -1126,16 +1126,16 @@ class _DefaultSemanticsGestureDelegate extends SemanticsGestureDelegate {
   @override
   void assignSemantics(RenderSemanticsGestureHandler renderObject) {
     assert(!detectorState.widget.excludeFromSemantics);
-    final Map<Type, GestureRecognizer> recognizers = detectorState._recognizers;
+    T getRecognizer<T extends GestureRecognizer>() => detectorState._recognizers[T] as T;
```

Comment:
This seems oddly complicated. Can just each _get*Handler do the cast?

Reply:
ok, I will revert this part and go with a simple explicit cast.